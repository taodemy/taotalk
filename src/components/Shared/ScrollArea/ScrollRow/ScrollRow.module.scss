@import '../../../../styles/color.scss';

.scrollRow {
  list-style: none;
  display: flex;
  align-items: center;
  gap: 12px;
  div:first-of-type {
    display: flex;
  }
}

.textContainer {
  display: flex;
  p {
    white-space: nowrap;
    font-size: 18px;
  }
  .scrollFontBlack {
    color: $black;
    padding-right: 4px;
  }
}

.player {
  padding: 10.62px 9.2px 9.2px 11.6px;
  background-color: $greyLight;
  border-radius: 4px;
  svg {
    color: $greyIcon;
    width: 19.2px;
    height: 20.18px;
  }
  &:hover {
    background-color: $cyanDark;
    svg {
      color: $white;
    }
  }
}

.playerActive {
  padding: 10.62px 9.2px 9.2px 11.6px;
  background-color: $cyanDark;
  border-radius: 4px;
  svg {
    color: $white;
    width: 19.2px;
    height: 20.18px;
  }
}
