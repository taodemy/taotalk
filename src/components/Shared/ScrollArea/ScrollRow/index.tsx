import React from 'react';
import styles from './ScrollRow.module.scss';
import IconButton from '../../Buttons/IconButton';

interface ScrollRowProps {
  itemOne: string;
  itemTwo: string;
  className?: string;
}
const ScrollRow: React.FC<ScrollRowProps> = ({ itemOne, itemTwo }) => {
  return (
    <div className={styles.scrollRow}>
      <IconButton iconTypes="player" />
      <div className={styles.textContainer}>
        <p className={`paragraph--p2 ${styles.scrollFontBlack}`}>{itemOne}</p>
        <p className={`paragraph--p2 ${styles.translation}`}>{itemTwo}</p>
      </div>
    </div>
  );
};

export default ScrollRow;
