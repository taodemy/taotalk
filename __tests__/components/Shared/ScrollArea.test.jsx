import { render, screen } from '@testing-library/react';
import ScrollRow from '../../../src/components/Shared/ScrollArea/ScrollRow';
import '@testing-library/jest-dom';
import Play from '../../../public/play.svg';

describe('render DirectionGuide component', () => {
  beforeEach(() => {
    render(<ScrollRow itemOne="lone" itemTwo="- solitaire" />);
  });

  it('renders a play image', () => {
    //Play svgr component in play button
    const playImage = render(<Play />);
    expect(playImage.firstChild).toMatchSnapshot();
  });

  it('renders an English Text', () => {
    const englishText = screen.getByText('lone');
    expect(englishText).toBeInTheDocument();
  });

  it('renders an Translation Text', () => {
    const translationText = screen.getByText('- solitaire');
    expect(translationText).toBeInTheDocument();
  });
});
