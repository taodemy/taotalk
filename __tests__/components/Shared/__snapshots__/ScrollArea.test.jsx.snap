// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render DirectionGuide component renders a play image 1`] = `undefined`;
